Makefile

```diff
@@ -21,21 +22,38 @@ endif
 all:
 	$(MSBUILD)
 
-prepare::
+prepare:: prepare-props
 	git submodule update --init --recursive
 	nuget restore
 	(cd external/Java.Interop && nuget restore)
+
+prepare-props:
 	cp Configuration.Java.Interop.Override.props external/Java.Interop/Configuration.Override.props
+	./build-tools/scripts/generate-os-info Configuration.OperatingSystem.props
```

Comment:
I left the `.sh` suffix by design. The idea is that one day we'll have the entire build working on Windows, with undoubtedly a number of changes, and the `Configuration.OperatingSystem.props` file will need to be generated there as well. Naming the Windows version of the script, say, `generate-os-info.cmd` will make it possible to use the same line as above, without anything special to make it work on Windows since Windows will look for the executable by appending the `.com`, `.exe`, `.bat`, `.cmd` extensions in some order to find the program to run.
It will be especially useful if we have to move the `make prepare` code to a special msbuild target or whatnot as we'll have a single line of code for all operating systems